devdoc/jdp/jdp-2019-06-ability-to-control-statements-execution.md

```diff
@@ -0,0 +1,101 @@
+# jdp-2019-06: Ability to control statements execution
+
+## Status
+
+- Draft
+
+## Type
+
+- Feature-Specification
+
+## Context
+
+At any given time, a different number of statements `org.firebirdsql.gds.ng.FbStatement` 
+can be executed in an application. The two main states in which a statement 
+can be exist are the execution `org.firebirdsql.gds.ngStatementState.EXECUTING` 
+and fetching `org.firebirdsql.gds.ngStatementState.CURSOR_OPEN`.
+
+The first problem is that these statuses allow us to track the status of the 
+statement, but do not allow it centrally for all statements. Partly, this problem 
+can be solved at the application level. But for this, each statement will have 
+to be wrapped to control executing. At the driver level, tracking is much easier.
+The second problem is the control of fetching rows. Executing or fetching data 
+can take considerable time and at the moment we do not know whether operations 
+are performed or not.
+
+The main idea of ​​this proposal is to add a mechanism that allows at an application
+level to know what operations are performed at a given moment in time.
+
+## Decision
+
+This proposal should be taken as a suggestion and may be revised.
+
+Implementation involves the addition of several interfaces and classes to control 
+the execution of statements.
+
+### Operation interface
+
+At the `org.firebirdsql.gds` package it is proposed to add the `Operation` interface, 
```

Comment:
Why in `org.firebirdsql.gds`, and not in `org.firebirdsql.gds.ng` like `StatementOperationAware`?

Reply:
Similar to 'EventHandler' which is in 'org.firebirdsql.gds'. Can be moved to another location.